### PR TITLE
add operations for attributes

### PR DESCRIPTION
**Pull request by nwlandry**
This adds additional operations for node/edge attributes.

### CODE REVIEW

**Comment by maximelucas** [comment]
Looks nice! Safer with a few tests?

File: xgi/classes/reportviews.py [hunk 1]
```diff
@@ -237,7 +237,24 @@ def filterby_attr(self, attr, val, mode="eq"):
         <https://github.com/ComplexGroupInteractions/xgi/blob/main/tutorials/Tutorial%206%20-%20Statistics.ipynb>`_.
```
Comment (leotrs) [inline xgi/classes/reportviews.py:238]: Notes
--------
Beware of using comparison modes ("lt", "gt", "leq", "geq") when the attribute is a string. For example, the string comparison `'10' < '9'` evaluates to `True`.

File: xgi/classes/reportviews.py [hunk 2]
```diff
@@ -237,7 +237,22 @@ def filterby_attr(self, attr, val, mode="eq"):
...
+        elif mode == "gt":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] > val]
+        elif mode == "leq":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] <= val]
+        elif mode == "geq":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] >= val]
+        elif mode == "between":
+            bunch = [idx for idx in self if val[0] <= self._id_attr[idx][attr] <= val[1]]
+        else:
+            raise ValueError(f"Unrecognized mode {mode}")
```
Comment (maximelucas) [inline xgi/classes/reportviews.py:257]: Would specify which modes are known:

raise ValueError(f"Unrecognized mode {mode}. mode must be one of ["eq", "neq", "lt", "gt", "leq", "geq", "between"].")

File: xgi/classes/reportviews.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/classes/reportviews.py:238]: The two ways I can think to deal with this are 1) exclude these IDs and 2) specify a default value.

[the remark this answers is not in the document]
Comment (nwlandry) [reply xgi/classes/reportviews.py:238]: I've implemented 1) for now.

File: xgi/classes/reportviews.py (hunk 2 above)
Comment (nwlandry) [reply xgi/classes/reportviews.py:257]: Good call. Fixed here and also in `filterby`

File: xgi/classes/reportviews.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (leotrs) [reply xgi/classes/reportviews.py:240]: We should handle it in the same way as `H.nodes.attrs` handles it, via the `missing` keyword. you don't have to implement it rn, but please open an issue :)

File: xgi/classes/reportviews.py [hunk 3]
```diff
@@ -225,36 +225,72 @@ def filterby(self, stat, val, mode="eq"):
...
             bunch = [node for node in self if val[0] <= values[node] <= val[1]]
         else:
-            raise ValueError(f"Unrecognized mode {mode}")
+            raise ValueError(
+                f"Unrecognized mode {mode}. mode must be one of 'eq', 'neq', 'lt', 'gt', 'leq', 'geq', or 'between'."
+            )
         return type(self).from_view(self, bunch)
 
-    def filterby_attr(self, attr, val, mode="eq"):
+    def filterby_attr(self, attr, val, mode="eq", default=None):
```
Comment (leotrs) [inline xgi/classes/reportviews.py:233]:
```suggestion
    def filterby_attr(self, attr, val, mode="eq", missing=None):
```

**Review by leotrs — APPROVED** [review]
Just change `default` to `missing` to it matches `H.nodes.attrs` (see `def attrs` in the `nodestats.py` module) and then you can merge.

File: xgi/classes/reportviews.py (hunk 3 above)
Comment (nwlandry) [reply xgi/classes/reportviews.py]: I liked your suggestion to match the interface so I implemented it now instead of adding it as an issue

**Pull request merged**